### PR TITLE
Use the signature types when building ABI info

### PR DESCRIPTION
**Pull request by SingleAccretion**
This switches `AddFinalArgsAndDetermineABIInfo` to exclusively use types from the signature, paving the way for us to remove the places where pessimizations and workarounds were needed to preserve the precise struct handles for arguments.

[No diffs](https://dev.azure.com/dnceng/public/_build/results?buildId=1820904&view=ms.vss-build-web.run-extensions-tab) as expected.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
`Build Linux arm64 Release NativeAOT` is a timeout (#70549).
`Installer Build and Test coreclr windows_x86 Debug` failed due to log file contention; I have seen this a few times in the past.

@dotnet/jit-contrib

**Review by jakobbotsch — APPROVED** [review]
LGTM, thanks for doing this!

File: src/coreclr/jit/morph.cpp [hunk 1]
```diff
@@ -3001,12 +3004,9 @@ void CallArgs::AddFinalArgsAndDetermineABIInfo(Compiler* comp, GenTreeCall* call
 #endif
         }
 
-        if (GlobalJitOptions::compFeatureHfa)
```
Comment (tannergooding) [inline src/coreclr/jit/morph.cpp:3004]: Are we relying on `varTypeIsValidHfaType` to report `false` if the feature isn't supported?

-- I'm also not overly familiar with the HFA handling. Does that also include HVA (that is ensuring `Vector128<T>`, which is `__m128`, and up to 4 sequential `Vector128<T>` are passed in register)?

**Review by tannergooding — APPROVED** [review]
Changes look good and mostly mechanical to me.

File: src/coreclr/jit/morph.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/morph.cpp:3004]: > Are we relying on `varTypeIsValidHfaType` to report false if the feature isn't supported?

Yep. It is "the pattern" for HFAs in the compiler.

> Does that also include HVA?

Also yes, for them the "HFA element type" will be `SIMD` (see `HfaTypeFromElemKind`).

**Pull request merged**